### ISSUE TITLE
From GMail to Office365 migration error (937 BAD Command Argument Error. 11)

### ISSUE THREAD

**Issue by xavier7179** [imapsync/imapsync#285]
Hi,

I'm using it on OSX to migrate from GMail to Office365 because technicians avoided the Office365 in place solution (I discover it but we do not have it enabled on our institution). I wanted to get one folder at the time, so I came up with this command line:

`mapsync --user1 <user1> --password1 <pwd1> --user2 <user2> --password2 <pwd2> --gmail1 --office2 --include "^FOLDER"`

It worked smoothly until one folder that is returning (<FOLDER> is to omit details for privacy):

`Err 1/1: - msg <FOLDER> {3540} could not append ( Subject:[=?TELETEX?Q?Fw=3ASome text before=5C=27Othertext?=], Date:["06-Dec-2006 18:01:37 +0000"], Size:[3307], Flags:[\Seen] ) to folder <FOLDER>: 937 BAD Command Argument Error. 11
Exiting with return value 111 (EXIT_WITH_ERRORS) 1/50 nb_errors/max_errors`

I also tried to solve it using the `reformine` tool (I know I'm under OSX but I built the tool using my own developed brew formulas) but still no luck. I looked around and seems no one reported this error code before.

**Comment by gilleslamiral** [comment]
The error message "BAD Command Argument Error. 11" is a well known one, but unfortunately not very relevant.

https://imapsync.lamiral.info/FAQ.d/FAQ.Office365.txt
...
````
Q. From XXX Office365 I get this error message sometimes: 
   "BAD Command Argument Error 11". What does it mean?

R. This error message comes from the Office365 IMAP server when it 
   encounters any problem. Most of the time it is one of the following:

...

  * Quota reached. The whole account is full.
    It can be upped by configuration.

...

  * Some messages have some lines too long. Use option --maxlinelength
    to skip messages whose max line length is over some bytes.
    --maxlinelength 1000 is an RFC2822 must but most servers support
    higher values. Office365 supports 10500 characters line length:

  imapsync ... --maxlinelength 10500

  In case you prefer fixing messages with long lines the hard way,
  instead of skipping them with --maxlinelength 10500, just use:

  On Windows
  imapsync ... --regexmess "s,(.{10500}),$1\r\n,g"

  On Unix
  imapsync ... --regexmess 's,(.{10500}),$1\r\n,g'
  
Have also in mind that Office365  closes the connection after 10 errors
encountered so you might also see "BYE Connection closed" errors from
Office365, which means Exchange leaves the session and says goodbye,
come back later. Redo some sync then.

On Linux, there is a good Python script in the tarball that can 
fix several things that Office365 has issues with.

Use it like this:

  ./imapsync ... --pipemess W/tools/fix_email_for_exchange.py

It often does some miracles on messages.
It's called fix_email_for_exchange because Office365 is Exchange,
or at least started to be an Exchange server with the same issues.
````

**Comment by xavier7179** [comment]
Thank you. Two things here:

1. the brew command is installed without those tools (I reported it on the homebrew discussion page...)
2. I tried all of them and the error is still there.

**Comment by gilleslamiral** [comment]
I have no clue.

There are so many ways for a message to not be RFC8222 compliant.
To go further add --debugcontent to dump the full content of the message, it's 3540 bytes long.
I can take a look at but I want it verbatim because every single byte count.

Another way is to try to transfer it with an other client like Thunderbird or Outlook and see if they succeed or fail the same way.

**Comment by xavier7179** [comment]
I believe I found the issue (btw the `--debugcontent` did not dump the message...). The Subject has a '\' char in it. I guess I need to play with `--regexmess` to remove those kinds of special chars, am I right?

**Comment by gilleslamiral** [comment]
> the --debugcontent did not dump the message

That's strange, the dump is delimited like this:
````
msg INBOX/22142 copying to INBOX ETA: mardi  8 juin 2021-06-08 10:38:40 +0200 CEST  2 s  1/1 msgs left
F message content begin next line (423 characters long)
Return-Path: <gilles@localhost.info>
Received: by lamiral.info (Postfix, from userid 1000)
        id BBBC25C1306; Tue,  8 Jun 2021 10:37:43 +0200 (CEST)
...
F message content ended on previous line
````

> The Subject has a '' char in it.

Ascii characters quotes and double-quotes are usually normal characters in a Subject header line. 
Did you finally found the issue?

**Comment by xavier7179** [comment]
No, I confirm that the option does not produce any dump like that.
Regarding the special char I did not notice that the commit of the message changed the character, it's a `\` in the object of the email. I tried with some mail program and it does not work either.

**Comment by gilleslamiral** [comment]
> Regarding the special char I did not notice that the commit of the message changed the character

So the message has finally been transferred?
If yes, then the error "could not append ... Command Argument Error. 11" doesn't show up anymore, right?

**Comment by xavier7179** [comment]
No, it does not work at all: either the imapsync or any actual mail program. The error is still the same but without the dump I cannot tell anything more.

**Comment by gilleslamiral** [comment]
Can you send me a --debugcontent logfile?

**Comment by xavier7179** [comment]
Sure! I'll drop you an email with that
